src/commands/dev/generate/flag.ts

```diff
@@ -0,0 +1,262 @@
+  'description',
+  'errors.InvalidDir',
+  'examples',
+  'flags.dry-run.summary',
+  'errors.FlagExists',
+  'summary',
+]);
+
+type Answers = {
+  char: Nullable<string>;
+  type: keyof typeof Flags;
+  name: string;
+  required: boolean;
+  multiple: boolean;
+  durationUnit: Lowercase<keyof typeof Duration.Unit>;
+  durationDefaultValue: number;
+  salesforceIdLength: '15' | '18' | 'None';
+  salesforceIdStartsWith: string;
+  fileOrDirExists: boolean;
+  integerMin: number;
+  integerMax: number;
+};
+
+const toLowerKebabCase = (str: string): string =>
+  str
+    .replace(/([a-z])([A-Z])/g, '$1-$2')
+    .replace(/[\s_]+/g, '-')
+    .toLowerCase();
+
+export default class DevGenerateFlag extends SfCommand<void> {
+  public static enableJsonFlag = false;
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  public static flags = {
+    'dry-run': Flags.boolean({
+      summary: messages.getMessage('flags.dry-run.summary'),
+      char: 'd',
+    }),
+  };
+
+  public async run(): Promise<void> {
+    const { flags } = await this.parse(DevGenerateFlag);
+
+    if (!fileExists('package.json')) throw messages.createError('errors.InvalidDir');
+
+    const ids = (await fg('src/commands/**/*.ts')).map((file) => {
+      const p = path.parse(file.replace(path.join('.', 'src', 'commands'), ''));
+      const topics = p.dir.split('/');
+      const command = p.name !== 'index' && p.name;
+      const id = [...topics, command].filter((f) => f).join(':');
+      return id === '' ? '.' : id;
+    });
+
+    const { command } = await this.prompt<{ command: string }>([
+      {
+        type: 'list',
+        name: 'command',
+        message: 'Which command is this for',
+        choices: ids.map((c) => toConfiguredId(c, this.config)),
+      },
+    ]);
+
+    const commandFilePath = path.join('.', 'src', 'commands', ...toStandardizedId(command, this.config).split(':'));
+
+    const existingFlags = await this.loadExistingFlags(commandFilePath);
+
+    const charToFlag = Object.entries(existingFlags).reduce((acc, [key, value]) => {
+      return value.char ? { ...acc, [value.char]: key } : acc;
+    }, {} as Record<string, string>);
+
+    const durationUnits = (Object.values(Duration.Unit).filter((unit) => typeof unit === 'string') as string[]).map(
+      (unit) => unit.toLowerCase()
+    );
+
+    const answers = await this.prompt<Answers>([
+      {
+        type: 'list',
+        name: 'type',
+        message: 'What type of flag is this',
+        choices: Object.keys(Flags).sort(),
+      },
+      {
+        type: 'input',
+        name: 'name',
+        message: 'What is the name of the flag',
+        validate: (input: string): string | boolean => {
+          if (toLowerKebabCase(input) !== input) {
+            return 'Flag name must be in kebab case';
```

Comment:
Is it worth having an example here? 🤔 `'Flag name must be in kebab case (example: my-flag)'`